### PR TITLE
MINOR: Docs on state store instantiation

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by bbejeck — APPROVED** [review]
Thanks @guozhangwang LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java [hunk 1]
```diff
@@ -46,7 +46,17 @@
     String name();
 
     /**
-     * Initializes this state store
+     * Initializes this state store.
+     *
+     * The implementation of this function must register the root store in the context via the
+     * {@link ProcessorContext#register(StateStore, StateRestoreCallback)} function, where the
+     * first {@code StateStore} parameter should always be the passed-in {@code root} object, and the
+     * second {@code StateRestoreCallback} is what users need to instantiate.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: `what users need to instantiate`

Unclear what this means.

File: streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java [hunk 2]
```diff
@@ -46,7 +46,17 @@
...
+     *
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: nit: add `<p>`

File: streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java [hunk 3]
```diff
@@ -46,7 +46,17 @@
...
+     * Note that if the state store implementation itself supports bulk writes, users can implement another
+     * interface {@link BatchingStateRestoreCallback} which extends {@link StateRestoreCallback} and could
+     * let users instantiate bulk-load restoration logic instead of restoring one record at a time.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: `let users instantiate` -- who is "users" ?

File: streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java [hunk 4]
```diff
@@ -46,7 +46,19 @@
     String name();
 
     /**
-     * Initializes this state store
+     * Initializes this state store.
+     *
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java]: If we want a new paragraph, we need to insert `<p>` here.

**Review by mjsax — APPROVED** [review]
LGTM.

One minor formatting nit. Feel free to merge.

**Pull request merged**